prepare_labeled_data_from_hunt_prestudy.py

```diff
@@ -0,0 +1,287 @@
+# coding: utf-8
+from __future__ import division, print_function
+
+import glob
+from time import time
+
+import pandas as pd
+
+from external_modules.detect_peaks import detect_peaks
+from tools.pandas_helpers import add_to_column_in_data_frame, subtract_from_column_in_data_frame, \
+    write_selected_columns_to_file
+
+
+def create_synchronized_back_and_thigh_file_for_subject(folder, s_id):
+    import subprocess
+
+    omconvert = "./private_data/conversion_scripts/omconvert/omconvert"
+    timesync = "./private_data/conversion_scripts/timesync/timesync"
+
+    back_cwa = glob.glob(folder + "/*_BACK_*" + s_id + ".cwa")[0]
+    thigh_cwa = glob.glob(folder + "/*_THIGH_*" + s_id + ".cwa")[0]
+
+    back_wav = folder + '/' + s_id + "_back.wav"
+    thigh_wav = folder + '/' + s_id + "_thigh.wav"
+
+    synchronized_csv = folder + '/' + s_id + "_thigh.resampled.csv"
+
+    # Create wav and CSV for back sensor
+    subprocess.call([omconvert, back_cwa, "-out", back_wav, "-csv-file", folder + '/' + s_id + "_back.csv"])
+
+    # Create wav for thigh sensor
+    subprocess.call([omconvert, thigh_cwa, "-out", thigh_wav])
+
+    # Synchronize them and make them a CSV
+    subprocess.call([timesync, back_wav, thigh_wav, "-csv", synchronized_csv])
+
+
+def find_claps_from_peaks(peak_array, required_claps=3, sampling_frequency=100, min_interval=0.15, max_interval=8.0):
+    claps = []
+    start_peak = peak_array[0]
+    previous_peak = start_peak
+    clap_count = 1
+    max_data_points_between_peaks = max_interval * sampling_frequency
+    min_data_points_between_peaks = min_interval * sampling_frequency
+
+    for peak in peak_array:
+        time_interval = peak - previous_peak
+
+        if min_data_points_between_peaks < time_interval <= max_data_points_between_peaks:
+            clap_count += 1
+        elif time_interval > max_data_points_between_peaks:
+            start_peak = peak
+            clap_count = 1  # Added resetting peak count when time interval too large from last peak. -- Eirik
+        else:
+            # Interval is less than minimum interval; still part of an already registered peak.
+            continue  # Skip to next loop iteration
```

Comment:
This depends on whether you want to count `time_interval` from the start of the previous peak or from the end of the previous peak. This is because by inserting the continue statement, the current line 58 is not executed.